### PR TITLE
Support parallel code execution in bandsv

### LINKED ISSUE #345: Support parallel code execution in bandsv

### PR DESCRIPTION
**Pull request by prin-r**
Fixes #345

### CODE REVIEW

File: chain/cmd/bandsv/main.go [hunk 1]
```diff
@@ -263,28 +264,62 @@ func handleExecute(c *gin.Context) {
...
+				return
+			}
+
+			qpi.answer = string(query)
+			qpi.httpStatus = http.StatusOK
+			chanQueryParallelInfo <- qpi
+		}(i, command)
+	}
+
+	var qpis []queryParallelInfo
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Change name

File: chain/cmd/bandsv/main.go [hunk 2]
```diff
@@ -263,28 +264,62 @@ func handleExecute(c *gin.Context) {
...
-			}, command.Arguments...),
-		}
-		query, err := execWithTimeout(dockerCommand, 10*time.Second)
-		if err != nil {
-			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	type queryParallelInfo struct {
+		index      int
+		answer     string
+		httpStatus int
+		err        interface{}
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Change to gin.H type

File: chain/cmd/bandsv/main.go [hunk 3]
```diff
@@ -263,28 +264,62 @@ func handleExecute(c *gin.Context) {
...
+	}
+	chanQueryParallelInfo := make(chan queryParallelInfo)
+	for i, command := range commands {
+		go func(index int, command Command) {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go:233]: This can leak: it is possible that some goroutines will stuck and waste resource forever

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandsv/main.go [hunk 4]
```diff
@@ -263,28 +264,62 @@ func handleExecute(c *gin.Context) {
...
+		if qpi.err != nil {
+			c.JSON(qpi.httpStatus, qpi.err)
 			return
 		}
-		answer = append(answer, string(query))
+		qpis = append(qpis, qpi)
 	}
 
-	b, _ := json.Marshal(answer)
+	sort.SliceStable(qpis, func(i, j int) bool {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: I don't think using stable sort is necessary here

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]
LGTM

**Review by taobun — APPROVED** [review]
LGTM

**Pull request merged**